In [1]:
import pandas as pd
import plotly.express as px

In [33]:
df = pd.read_csv('simulations_day_output.csv', sep=';', header=None)
df.columns = ['simulation_key', 'missing_plan_cnt_first_delay', 'missing_plan_cnt_planned_time', 'whole_process_delay']
df['controller_name'] = df.simulation_key.str.split('-', expand=True)[0]
df['break_strategy'] = df.simulation_key.str.split('-', expand=True)[1]
df['simulation_index'] = df.simulation_key.str.split('-', expand=True)[2].astype(int)
df['produced_in_planned_time'] = 1416 - df.missing_plan_cnt_planned_time
df['whole_process_delay'] = df.whole_process_delay.str.replace(',', '.').astype(float)
df['process_delay_minutes'] = df.whole_process_delay / 60
df['process_delay_hours'] = df.process_delay_minutes / 60
df.head()

simulation_key  missing_plan_cnt_first_delay  \
0  naive-break_skip-1                             0   
1  naive-break_skip-2                             0   
2  naive-break_skip-3                          1391   
3  naive-break_skip-4                          1415   
4  naive-break_skip-5                          1386   

   missing_plan_cnt_planned_time  whole_process_delay controller_name  \
0                              0                 0.00           naive   
1                              0                 0.00           naive   
2                            342             25687.35           naive   
3                            341             25651.99           naive   
4                            290             20776.65           naive   

  break_strategy  simulation_index  produced_in_planned_time  \
0     break_skip                 1                      1416   
1     break_skip                 2                      1416   
2     break_skip                 3                      1074   
3     break_skip                 4                      1075   
4     break_skip                 5                      1126   

   process_delay_minutes  process_delay_hours  
0               0.000000             0.000000  
1               0.000000             0.000000  
2             428.122500             7.135375  
3             427.533167             7.125553  
4             346.277500             5.771292

In [35]:
df.describe()

missing_plan_cnt_first_delay  missing_plan_cnt_planned_time  \
count                   1200.000000                    1200.000000   
mean                    1078.953333                      24.125833   
std                      518.697658                      75.428661   
min                        0.000000                       0.000000   
25%                     1083.750000                       0.000000   
50%                     1362.000000                       0.000000   
75%                     1382.000000                       5.000000   
max                     1415.000000                     343.000000   

       whole_process_delay  simulation_index  produced_in_planned_time  \
count          1200.000000       1200.000000               1200.000000   
mean           1794.164483        600.500000               1391.874167   
std            5468.415229        346.554469                 75.428661   
min               0.000000          1.000000               1073.000000   
25%               7.380000        300.750000               1411.000000   
50%             110.060000        600.500000               1416.000000   
75%             504.625000        900.250000               1416.000000   
max           25806.420000       1200.000000               1416.000000   

       process_delay_minutes  process_delay_hours  
count            1200.000000          1200.000000  
mean               29.902741             0.498379  
std                91.140254             1.519004  
min                 0.000000             0.000000  
25%                 0.123000             0.002050  
50%                 1.834333             0.030572  
75%                 8.410417             0.140174  
max               430.107000             7.168450

In [40]:
df[df.missing_plan_cnt_planned_time == 0].shape[0] / df.shape[0]

0.6083333333333333

In [47]:
df[df.whole_process_delay == 0].shape[0] / df.shape[0]

0.16083333333333333

In [48]:
df[(df.whole_process_delay != 0) & (df.missing_plan_cnt_planned_time == 0)]

simulation_key  missing_plan_cnt_first_delay  \
29      naive-break_skip-30                          1368   
30      naive-break_skip-31                          1380   
34      naive-break_skip-35                          1355   
35      naive-break_skip-36                          1372   
39      naive-break_skip-40                          1351   
40      naive-break_skip-41                          1375   
41      naive-break_skip-42                          1353   
42      naive-break_skip-43                          1254   
43      naive-break_skip-44                          1357   
44      naive-break_skip-45                          1334   
45      naive-break_skip-46                          1368   
46      naive-break_skip-47                          1370   
47      naive-break_skip-48                          1332   
48      naive-break_skip-49                          1376   
49      naive-break_skip-50                          1334   
50      naive-break_skip-51                          1350   
51      naive-break_skip-52                          1362   
52      naive-break_skip-53                          1282   
53      naive-break_skip-54                           863   
54      naive-break_skip-55                          1334   
55      naive-break_skip-56                          1348   
56      naive-break_skip-57                          1415   
57      naive-break_skip-58                           928   
58      naive-break_skip-59                          1224   
61      naive-break_skip-62                          1415   
62      naive-break_skip-63                            42   
64      naive-break_skip-65                           667   
65      naive-break_skip-66                          1316   
66      naive-break_skip-67                          1257   
68      naive-break_skip-69                          1024   
...                     ...                           ...   
1142  naive-break_skip-1143                          1354   
1143  naive-break_skip-1144                          1377   
1144  naive-break_skip-1145                          1370   
1145  naive-break_skip-1146                          1371   
1146  naive-break_skip-1147                          1335   
1147  naive-break_skip-1148                          1012   
1148  naive-break_skip-1149                          1030   
1149  naive-break_skip-1150                          1240   
1150  naive-break_skip-1151                          1255   
1151  naive-break_skip-1152                          1140   
1152  naive-break_skip-1153                          1363   
1153  naive-break_skip-1154                          1206   
1155  naive-break_skip-1156                          1351   
1156  naive-break_skip-1157                          1315   
1157  naive-break_skip-1158                          1356   
1158  naive-break_skip-1159                          1352   
1159  naive-break_skip-1160                            94   
1163  naive-break_skip-1164                           919   
1164  naive-break_skip-1165                           812   
1173  naive-break_skip-1174                           860   
1174  naive-break_skip-1175                          1415   
1183  naive-break_skip-1184                           185   
1184  naive-break_skip-1185                          1271   
1189  naive-break_skip-1190                          1353   
1190  naive-break_skip-1191                          1347   
1191  naive-break_skip-1192                          1366   
1193  naive-break_skip-1194                          1365   
1194  naive-break_skip-1195                          1254   
1195  naive-break_skip-1196                          1403   
1199  naive-break_skip-1200                          1388   

      missing_plan_cnt_planned_time  whole_process_delay controller_name  \
29                                0               184.76           naive   
30                                0               179.92           naive   
34

In [45]:
df[['produced_in_planned_time', 'process_delay_minutes']].describe()

produced_in_planned_time  process_delay_minutes
count               1200.000000            1200.000000
mean                1391.874167              29.902741
std                   75.428661              91.140254
min                 1073.000000               0.000000
25%                 1411.000000               0.123000
50%                 1416.000000               1.834333
75%                 1416.000000               8.410417
max                 1416.000000             430.107000

In [43]:
df[['produced_in_planned_time', 'process_delay_minutes']].mean()

produced_in_planned_time    1391.874167
process_delay_minutes         29.902741
dtype: float64

| Experiment | Plánovaný počet kusů | Procentuální počet splněných plánů | Procentuální počet plánů bez zpoždění | Průměrný počet vyprodukovaných kusů za plánované období 1-|
| --- | --- | --- | --- | --- |
| Naivní bez přeskládávání | 1416 | .608 | 0.161 | 1391.87 | 29.90

###  Plan je 1416

In [37]:
df.produced_in_planned_time.describe(percentiles=[0, .1, .2, .3, .4, .5, .6, .7, .8, .9, 1])

count    1200.000000
mean     1391.874167
std        75.428661
min      1073.000000
0%       1073.000000
10%      1390.000000
20%      1408.000000
30%      1413.000000
40%      1416.000000
50%      1416.000000
60%      1416.000000
70%      1416.000000
80%      1416.000000
90%      1416.000000
100%     1416.000000
max      1416.000000
Name: produced_in_planned_time, dtype: float64

In [34]:
px.box(df, y='produced_in_planned_time', color='controller_name')

In [13]:
df_grouped_zeros = pd.DataFrame(index=df.controller_name.unique(), columns=['zeros_count', 'count'])
for group_name, df_group in df.groupby('controller_name'):
    df_grouped_zeros.loc[group_name, 'zeros_count'] = df_group.missing_plan_cnt_first_delay.isin([0]).sum()
    df_grouped_zeros.loc[group_name, 'count'] = df_group.shape[0]
df_grouped_zeros = df_grouped_zeros.reset_index()
df_grouped_zeros

index  zeros_count  count
0  naive          193   1200

In [20]:
px.box(df, y='process_delay_minutes', color='controller_name')

Maximalni delay uspesne vyrobene cele vyrobni rady

In [30]:
df[(df.missing_plan_cnt_planned_time == 0)].sort_values('whole_process_delay', ascending=False).head()

simulation_key  missing_plan_cnt_first_delay  \
805    naive-break_skip-806                          1351   
35      naive-break_skip-36                          1372   
395    naive-break_skip-396                          1377   
1075  naive-break_skip-1076                          1398   
974    naive-break_skip-975                          1358   

      missing_plan_cnt_planned_time  whole_process_delay controller_name  \
805                               0               293.42           naive   
35                                0               280.17           naive   
395                               0               274.35           naive   
1075                              0               266.47           naive   
974                               0               263.61           naive   

     break_strategy  simulation_index  process_delay_minutes  \
805      break_skip               806               4.890333   
35       break_skip                36               4.669500   
395      break_skip               396               4.572500   
1075     break_skip              1076               4.441167   
974      break_skip               975               4.393500   

      process_delay_hours  
805              0.081506  
35               0.077825  
395              0.076208  
1075             0.074019  
974              0.073225

In [23]:
df[(df.missing_plan_cnt_planned_time == 0)].shape

(730, 9)

In [26]:
px.box(df[(df.missing_plan_cnt_planned_time == 0)], y='whole_process_delay')

In [46]:
px.box(df, y='missing_plan_cnt_first_delay')